src/Elastic.Apm/Model/Payload/Span.cs

```diff
@@ -40,49 +50,90 @@ public Span(string name, string type, Transaction transaction)
 		/// <value>The duration.</value>
 		public double? Duration { get; set; }
 
-		public int Id { get; set; }
+		public string Id { get; set; }
 
-		private string _name;
 		public string Name
 		{
 			get => _name;
 			set
 			{
 				if (value.Length > Consts.PropertyMaxLength)
-					value = $"{value.Substring(0, Consts.PropertyMaxLength-3)}...";
+					value = $"{value.Substring(0, Consts.PropertyMaxLength - 3)}...";
 				_name = value;
 			}
 		}
 
-		[JsonProperty("Stacktrace")]
-		public List<Stacktrace> StackTrace { get; set; }
+		[JsonProperty("parent_id")]
+		public string ParentId { get; set; }
 
-		public decimal Start { get; set; }
+		[JsonProperty("Stacktrace")]
+		public List<StackFrame> StackTrace { get; set; }
 
 		public string Subtype { get; set; }
 
 		[JsonIgnore]
 		public Dictionary<string, string> Tags => Context.Tags;
 
-		internal Transaction Transaction;
+		//public decimal Start { get; set; }
+		public long Timestamp => _start.ToUnixTimeMilliseconds() * 1000;
+
+		[JsonProperty("trace_id")]
+		public string TraceId { get; set; }
 
-		public Guid TransactionId => Transaction.Id;
+		[JsonProperty("Transaction_id")]
```

Comment:
according to shema it should be transaction_id :)

Reply:
👍 It was fixed by `CamelCasePropertyNamesContractResolver`, so it worked with capital `T`, but I agree that we should use what the schema defines here. I adapted all `JsonProperty`s accordingly.